internal/helm/getter/client_opts.go

```diff
@@ -139,9 +139,7 @@ func GetClientOpts(ctx context.Context, c client.Client, obj *sourcev1.HelmRepos
 		if authErr != nil {
 			return nil, "", fmt.Errorf("failed to get credential from '%s': %w", obj.Spec.Provider, authErr)
 		}
-		if authenticator != nil {
```

Comment:
This is breaking the tests big time because they don't have the default (`generic`). To fix this I think we can change the `else if` right above to:

```go
} else if p := obj.Spec.Provider; p != "" && p != sourcev1.GenericOCIProvider && obj.Spec.Type == sourcev1.HelmRepositoryTypeOCI && ociRepo {
```

Reply:
Thank you for the review!

I've fixed the provider check in 3e821693 to exclude empty strings as suggested.

The tests should now pass.

I apologize for not running the full test suite locally before submitting the PR. I initially relied on CI to catch any issues but
unfortunately had to step away before seeing the results. I should have checked the CI outcome before the PR was reviewed.

Once the tests pass and you're satisfied with the changes, I'll squash the fixup commit.